manager/integration/framework/node_exec/local_cluster.py

```diff
@@ -0,0 +1,17 @@
+from node_exec.abstract_cloud_provider import AbstractCloudProvider
+
+class LocalCluster(AbstractCloudProvider):
+    def __init__(self) -> None:
+        super().__init__()
+    
+    def get_all_node_instances(self):
+        raise NotImplemented
+    
+    def get_node_instance(self, node_name):
+        raise NotImplemented
+        
+    def power_off_node_instance(self, node_name):
+        raise NotImplemented
+    
+    def power_on_node_instance(self, node_name=""):
```

Comment:
If in some cases it's not possible to (determine how to) `power on/off` an instance, I think we can just only support `power on/off` test case only on AWS, and skip them on other environments, since `power on/off` is the only problematic operation, other operations can all be implemented using SSH. And then we can get rid of this `AbstractCloudProvider` and make things much simpler.

WDYT @innobead

Reply:
I updated the power on/off function to throw an exception when running on a local cluster. The power on/off functionality are only implemented when running the cluster on AWS.

https://github.com/longhorn/longhorn-tests/blob/15ca9dcba0534243fb68b9d0060173eafd129a5f/e2e/libs/node/operations/local_cluster.py#L20-L26